apps/quicklaunch/app.js

```diff
@@ -8,44 +8,58 @@
   };
 
   let swipeHandler = (lr,ud) => {
-    if (lr == -1 && settings.extleftapp && settings.extleftapp.src){ if (settings.extleftapp.name == "Show Launcher") Bangle.showLauncher(); else if (!storage.read(settings.extleftapp.src)) reset("extleftapp"); else load(settings.extleftapp.src); }
-    if (lr == 1 && settings.extrightapp && settings.extrightapp.src){ if (settings.extrightapp.name == "Show Launcher") Bangle.showLauncher(); else if (!storage.read(settings.extrightapp.src)) reset("extrightapp"); else load(settings.extrightapp.src); }
-    if (ud == -1 && settings.extupapp && settings.extupapp.src){ if (settings.extupapp.name == "Show Launcher") Bangle.showLauncher(); else if (!storage.read(settings.extupapp.src)) reset("extupapp"); else load(settings.extupapp.src); }
-    if (ud == 1 && settings.extdownapp && settings.extdownapp.src){ if (settings.extdownapp.name == "Show Launcher") Bangle.showLauncher(); else if (!storage.read(settings.extdownapp.src)) reset("extdownapp"); else load(settings.extdownapp.src); }
+    if (lr == -1) trace = leaveTrace(trace+"l"); // l=left, 
+    if (lr == 1) trace = leaveTrace(trace+"r"); // r=right,
+    if (ud == -1) trace = leaveTrace(trace+"u"); // u=up,
+    if (ud == 1) trace = leaveTrace(trace+"d"); // d=down.
+    if (lr == -1 && settings[trace+"app"] && settings[trace+"app"].src){ if (settings[trace+"app"].name == "Show Launcher") Bangle.showLauncher(); else if (!storage.read(settings[trace+"app"].src)) reset(trace+"app"); else load(settings[trace+"app"].src); }
+    if (lr == 1 && settings[trace+"app"] && settings[trace+"app"].src){ if (settings[trace+"app"].name == "Show Launcher") Bangle.showLauncher(); else if (!storage.read(settings[trace+"app"].src)) reset(trace+"app"); else load(settings[trace+"app"].src); }
+    if (ud == -1 && settings[trace+"app"] && settings[trace+"app"].src){ if (settings[trace+"app"].name == "Show Launcher") Bangle.showLauncher(); else if (!storage.read(settings[trace+"app"].src)) reset(trace+"app"); else load(settings[trace+"app"].src); }
+    if (ud == 1 && settings[trace+"app"] && settings[trace+"app"].src){ if (settings[trace+"app"].name == "Show Launcher") Bangle.showLauncher(); else if (!storage.read(settings[trace+"app"].src)) reset(trace+"app"); else load(settings[trace+"app"].src); }
   };
 
-  let removeUI = ()=>{if (timeoutToClock) clearTimeout(timeoutToClock);}
-
   Bangle.setUI({
     mode: "custom",
     touch: touchHandler,
     swipe : swipeHandler,
-    remove: removeUI // Compatability with Fastload Utils.
+    remove: ()=>{if (timeoutToClock) clearTimeout(timeoutToClock);} // Compatibility with Fastload Utils.
   });
 
   g.clearRect(Bangle.appRect);
-  Bangle.loadWidgets(); // Compatability with Fastload Utils.
+  let loadWidgets = "Bangle.loadWidgets();"; // Hack: Fool Fastload Utils that we call Bangle.loadWidgets(). This way we get the fastest possibe loading in whichever environment we find ourselvs.
```

Comment:
@halemmerich is this smart or dumb?

Reply:
I suspect in case of slow loading you don't want to load the widgets at all and just use the full screen to save on loading? In this case faking widget handling will probably work fine. You could just use a string without storing it in a variable though, just like "ram" for example.

Reply:
Yes that's precisely it. Ok, thanks!